tests/tests/write_texture.rs

```diff
@@ -90,107 +122,119 @@ fn write_texture_subset_2d() {
+        for byte in &data[..(total_size_in_bytes as usize)] {
+            assert_eq!(*byte, val);
         }
-        for byte in &data[(size as usize * 2)..] {
+        for byte in &data[(total_size_in_bytes as usize)..] {
             assert_eq!(*byte, 0);
         }
     });
 }
 
 #[test]
 #[wasm_bindgen_test]
-fn write_texture_subset_3d() {
-    let size = 256;
-    let depth = 4;
-    let parameters = TestParameters::default();
-    initialize_test(parameters, |ctx| {
-        let tex = ctx.device.create_texture(&wgpu::TextureDescriptor {
-            label: None,
-            dimension: wgpu::TextureDimension::D3,
-            size: wgpu::Extent3d {
-                width: size,
-                height: size,
-                depth_or_array_layers: depth,
-            },
-            format: wgpu::TextureFormat::R8Uint,
-            usage: wgpu::TextureUsages::COPY_DST
-                | wgpu::TextureUsages::COPY_SRC
-                | wgpu::TextureUsages::TEXTURE_BINDING,
-            mip_level_count: 1,
-            sample_count: 1,
-            view_formats: &[],
-        });
-        let data = vec![1u8; (size * size) as usize * 2];
-        // Write the first two slices
-        ctx.queue.write_texture(
-            wgpu::ImageCopyTexture {
-                texture: &tex,
-                mip_level: 0,
-                origin: wgpu::Origin3d::ZERO,
-                aspect: wgpu::TextureAspect::All,
-            },
-            bytemuck::cast_slice(&data),
-            wgpu::ImageDataLayout {
-                offset: 0,
-                bytes_per_row: Some(size),
-                rows_per_image: Some(size),
-            },
-            wgpu::Extent3d {
-                width: size,
-                height: size,
-                depth_or_array_layers: 2,
-            },
-        );
-
-        ctx.queue.submit(None);
-
-        let read_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
-            label: None,
-            size: (size * size * depth) as u64,
-            usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::COPY_DST,
-            mapped_at_creation: false,
-        });
-
-        let mut encoder = ctx
-            .device
-            .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+fn write_texture_subset_2d() {
+    let format = wgpu::TextureFormat::R8Uint;
+    let mips_count = 1;
+    let tex_size = wgpu::Extent3d {
+        width: 256,
+        height: 256,
+        depth_or_array_layers: 1,
+    };
+
+    let bytes_per_row = tex_size.width;
+    let rows_per_image = tex_size.height;
+
+    let total_bytes_in_copy = total_bytes_in_copy(format, bytes_per_row, rows_per_image, tex_size);
+
+    test(
+        format,
+        LayoutDesc {
+            bytes_per_row,
+            rows_per_image,
+            total_size_in_bytes: total_bytes_in_copy,
+        },
+        0,
+        mips_count,
+        tex_size,
+        tex_size,
+        tex_size,
```

Comment:
This test seems to now copy the whole texture instead of a subset (prev. `height: 2`).